x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/index/store/SearchableSnapshotDirectory.java

```diff
@@ -393,33 +403,64 @@ protected void doRun() throws Exception {
                                         TimeValue.timeValueNanos(statsCurrentTimeNanosSupplier.getAsLong() - startTimeInNanos).millis()
                                     )
                                 );
+                            }));
+                        }
+                        enqueued = true;
+                    }
+                } catch (Exception e) {
```

Comment:
This never rethrows the exception?

Reply:
If I understand correctly rethrowing here would fail the shard which I think it's something we should try to avoid in case of prewarming. Maybe we should rethrow only non-AlreadyClosedException, so that we don't fail the shard if the node is closing.

Reply:
Something like 92bcbcb85f3a1ca4a2ad7081dd0c012712f06df0